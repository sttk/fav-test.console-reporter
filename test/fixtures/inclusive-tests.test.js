'use strict';

function check_test_environment() {
  return false;
}

describe('Array (1)', function() {
  describe.skip('#indexOf()', function() {
    it('should return -1 unless present', function() {
      // this test will not be run
    });

    it('should return the index when present', function() {
      // this test will be run
    });
  });
});

describe('Array (2)', function() {
  describe('#indexOf()', function() {
    it.skip('should return -1 unless present', function() {
      // this test will not be run
    });

    it('should return the index when present', function() {
      // this test will be run
    });
  });
});

it('should only test in the correct environment (1)', function() {
  if (check_test_environment()) {
    // make assertions
  } else {
    this.skip();
  }
});

it('should only test in the correct environment (2)', function() {
  if (check_test_environment()) {
    // make assertions
  } else {
    // do nothing
  }
});

describe('outer', function () {
  before(function () {
    this.skip();
  });

  after(function () {
    console.log('after in "outer" will be executed');
  });

  describe('inner', function () {
    before(function () {
      console.log('before in "innter" will be skipped');
    });

    after(function () {
      console.log('after in "innter" will be skipped');
    });

    it('Run test in "inner"', function() {
    });
  });

  it('Run test in "outer"', function() {
  });
});

