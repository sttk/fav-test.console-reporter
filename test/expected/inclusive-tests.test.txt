

  Array (1)
    #indexOf()
      - should return -1 unless present
      - should return the index when present

  Array (2)
    #indexOf()
      - should return -1 unless present
      ✓ should return the index when present

  - should only test in the correct environment (1)
  ✓ should only test in the correct environment (2)

  outer
    inner
      - Run test in "inner"
    - Run test in "outer"
after in "outer" will be executed


  2 passing (5ms)
  6 pending

